worker/download.go

```diff
@@ -213,9 +209,16 @@ func (mgr *downloadManager) DownloadObject(ctx context.Context, w io.Writer, o o
 	nextSlabChan := make(chan struct{}, 1)
 	nextSlabChan <- struct{}{}
 
-	// launch a goroutine to launch consecutive slab downloads
+	// create a response channel
 	responseChan := make(chan *slabDownloadResponse)
 	defer close(responseChan)
+
+	// ensure everything cancels if download is done, make sure to cancel the
+	// context before closing the channel to avoid sending down a closed channel
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I don't think this is a guarantee that we aren't sending down a closed channel if the timing is really unfortunate?

Reply:
Will make some changes, I thought about this but figured it was so unlikely to happen we probably want to avoid the added complexity but it's not that bad it seems to introduce a waitgroup. I did have to add something funky to take care of cleaning up the sector response chan but you'll find out in review.